### ISSUE TITLE
[Bug]: Camera.onUserTrackingModeChange not called on followUserMode change on Android 

### ISSUE THREAD

**Issue by andrei-tofan** [rnmapbox/maps#3017]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### Platform

Android

### `@rnmapbox/maps` version

10.0.12

### Standalone component to reproduce

```typescript
import React, {useState} from 'react';
import Mapbox, {Camera, MapView, UserTrackingMode} from '@rnmapbox/maps';
import {Button, Platform} from 'react-native';

const App = () => {
  const [followUserMode, setFollowUserMode] = useState<
    UserTrackingMode | undefined
  >(UserTrackingMode.Follow);

  return (
    <>
      {Object.values(UserTrackingMode).map((mode: any) => (
        <Button
          key={mode}
          title={`${mode} ${mode === followUserMode ? '(active)' : ''}`}
          onPress={() => {
            console.log('setMode', mode);
           setFollowUserMode(mode);
          }}
        />
      ))}
      <MapView style={{flex: 1}}>
        <Mapbox.UserLocation visible showsUserHeadingIndicator />

        <Camera
          followUserLocation
          followUserMode={followUserMode}
          onUserTrackingModeChange={(event: any) => {
            console.log(
              `onUserTrackingModeChange(${event.nativeEvent.payload.followUserMode})`,
            );

            setFollowUserMode(
              event.nativeEvent.payload.followUserMode || undefined,
            );
          }}
        />
      </MapView>
    </>
  );
};

export default App;

```

### Observed behavior and steps to reproduce

When changing the `followUserMode`, the `onUserTrackingModeChange` callback is called on iOS but is not called on Android.

### Expected behavior

The `onUserTrackingModeChange` callback should be called on both platforms.

### Additional links and references

Found a comment about this issue here:
https://github.com/rnmapbox/maps/issues/1090

**Comment by mfazekas** [comment]
@andrei-tofan sorry I did not have time to look into it.

 Trying on current main or 10.1 beta-s would help.

You can also try to debug the viewport code https://github.com/rnmapbox/maps/pull/2767/files

**Comment by mfazekas** [comment]
The button's were changed so it got called. 

This is the console log:

```js
 LOG  Running "RNMapboxGLExample" with {"rootTag":21}
 LOG  onUserTrackingModeChange([object Object])
 LOG  onUserTrackingModeChange([object Object])
 LOG  mode normal
 LOG  onUserTrackingModeChange([object Object])
 LOG  onUserTrackingModeChange([object Object])
 LOG  mode compass
 LOG  onUserTrackingModeChange([object Object])
 LOG  onUserTrackingModeChange([object Object])
 LOG  mode normal
 LOG  onUserTrackingModeChange([object Object])
 LOG  onUserTrackingModeChange([object Object])
 LOG  mode compass
 LOG  onUserTrackingModeChange([object Object])
 LOG  onUserTrackingModeChange([object Object])
```

**Comment by jaltin** [comment]
Hi @mfazekas,

I am a colleague of @andrei-tofan and helping him with this issue, hence why I also get involved in this thread.

We are not getting the onUserTrackingModeChange log in the same way as you on Android.

See the attached video where I try to illustrate when it is firing only when dragging on the map with finger (an android phone running os 9).

https://github.com/rnmapbox/maps/assets/2103778/c3af00a7-5b3a-4537-a58a-743d4aa9a4b6

I have also tested on an Android tablet with os 11 and having the same problem there.

On iOS it is firing on every mode change, just as in your console log example.

I hope this helps in illustrating the problem we are having, but let me know if we can provide further info.

**Comment by mfazekas** [comment]
@jaltin Ok sorry I wasn't careful enough reading the issue. So you're missing `onUserTrackingModeChange` in case you're programatically changing followUserMode on the `Camera`.

**Comment by mfazekas** [comment]
@andrei-tofan this should be addressed in [10.1.0-beta.13](https://www.npmjs.com/package/@rnmapbox/maps/v/10.1.0-beta.13)
